### ISSUE TITLE
LeaveDuringTurn doesn't work as documented

### ISSUE THREAD

**Issue by habitoti** [playgameservices/play-games-plugin-for-unity#2561]
According to the documentation, an automatched player could leave the match before taking a turn and return the match to state "Automatching", letting others join in.

> In general, when a participant leaves a match, another player cannot join and take that participant’s place. One exception is when a player who joined by auto-match and who has not taken a turn calls TurnBasedMultiplayerClient.leaveMatchDuringTurn(). In this case, the match reverts to the MATCH_STATUS_AUTO_MATCHING state and another player can take over the place of the participant who left.

However, this doesn't work. When an automatched player calls LeaveDuringTurn before sending his turn, the match will be cancelled.

This leads to a very bad user experience. We permanently see "newbies" join via quick match into automatches and seem to be "shocked" they ended up in a running game and bail out right away w/o taking a turn. This would be no problem if the seat would be released for someone else to join in, however instead the whole match is cancelled!

EDIT: it will cancel the match if only the initiating player is left in the match. If there are already other players, it actually just releases the slot for others to join in. I have no clue what the reasoning would be to cancel the match in the other case. It is just annoying...

**Comment by habitoti** [comment]
@olehkuznetsov:  some more research for your records ;-)

- If a player is matched into a quick match setup when he did an explicit invitation with an automatch slot, he can leave before taking a turn, and then the slot is properly released w/o canceling the quick match. 

- If however he also started via quick matched and is joined in as the first additional participant and then leaves w/o a turn, that initial match is cancelled. (SHOULD NOT)

- If second player quick matches like above, but the target match has already 2 or more participants, the automatch slot is properly released (w/o cancelling that match) when he leaves.

So bottom line: the inconsistent behavior happens when two „fresh“ quick matches are joined and the new player leaves his turn.

**Comment by habitoti** [comment]
@olehkuznetsov : What would be the best channel to discuss other (peculiar ;-) and probably not plugin-related observations around GPGS (esp. quick matching)? Is there a GPGS forum that would fit for technical discussions and that you guys follow? There is a Google Play community, but it seems focussed on consumption by end-users rather than technical issues/topics.

**Comment by olehkuznetsov** [comment]
Thanks for your question.  Play Games Services multiplayer system has been deprecated (see this notice), so we are closing this issue
